### ISSUE TITLE
Recompiling entire project after a single error

### ISSUE THREAD

**Issue by parsonsmatt** [urbanslug/yesod-devel#19]
Currently, `wai-devel` will recompile the entire application if a single error occurs during compilation.

`ghci` is able to keep the parts of the application that didn't have errors, and pick up where it leaves off. Is there a way to get `wai-devel` to do the same thing?

**Comment by urbanslug** [comment]
How long has this been happening because it usually recompiles only the changes. Anyway I will look into it.

**Comment by parsonsmatt** [comment]
Yeah, if there is just a change without errors, it recompiles only the modules necessary. If there's a compile-time error (eg syntax error, type error), then it has to recompile the whole project.

Thanks!

**Comment by urbanslug** [comment]
Oh that's what you mean. Yeah that's what it does. I actually thought that in the case of an error GHC usually recompiles the entire project. I didn't know what you are suggesting was doable. I will look into it and see what I can do.
